consul/client.go

```diff
@@ -328,9 +369,26 @@ func (c *Client) localEvent(event serf.UserEvent) {
 
 // RPC is used to forward an RPC call to a consul server, or fail if no servers
 func (c *Client) RPC(method string, args interface{}, reply interface{}) error {
-	// Check the last rpc time
+	// Check to make sure we haven't spent too much time querying a
+	// single server
+	now := time.Now()
+	if !c.connRebalanceTime.IsZero() && now.After(c.connRebalanceTime) {
+		c.logger.Printf("[DEBUG] consul: connection time to server %s exceeded, rotating server connection", c.lastServer.Addr)
+		c.lastServer = nil
+	}
+
+	// Allocate these vars on the stack before the goto
```

Comment:
I had the exact same thought, but the compiler doesn't reevaluate the scope of the variables on the stack.  I almost rejiggered things to use a block and scope the variables but decided that was unnecessary babysitting of the language and left it with just a comment.  It's almost to the point where this function should be decomposed into two separate functions.